airflow/utils/log/file_task_handler.py

```diff
@@ -80,6 +80,35 @@ def _render_filename(self, ti, try_number):
                                              task_id=ti.task_id,
                                              execution_date=ti.execution_date.isoformat(),
                                              try_number=try_number)
+    from airflow.utils.db import provide_session
```

Comment:
Let's put all `import`s at the beginning part of the file, as suggested in PEP 8 (https://www.python.org/dev/peps/pep-0008/)

Reply:
The same for your line `from airflow import models` and `import json`